src/components/PropertyCheckbox.tsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { useCompanyChanges } from '$app/common/hooks/useCompanyChanges';
+import { useHandleCurrentCompanyChangeProperty } from '$app/pages/settings/common/hooks/useHandleCurrentCompanyChange';
+import { ReactElement, useEffect, useState } from 'react';
+import { Checkbox } from './forms';
+import { cloneDeep } from 'lodash';
+import { Settings } from '$app/common/interfaces/company.interface';
+import { useCurrentSettingsLevel } from '$app/common/hooks/useCurrentSettingsLevel';
+import classNames from 'classnames';
+
+type DefaultValueType = string | boolean | number;
+
+interface Props {
+  propertyKey: keyof Settings;
+  labelElement: ReactElement;
+  defaultValue?: DefaultValueType;
+  onCheckboxChange?: (value: boolean) => void;
+  checked?: boolean;
+}
+
+export function PropertyCheckbox(props: Props) {
```

Comment:
@beganovich I think I have fairly good reasons for choosing to use a component instead of a custom hook. First, I'd like to share the difference in code volume without using these two components (`PropertyCheckbox` and `SettingsLabel`). When using these components, the code looks like this:

```
leftSide={
          <PropertyCheckbox
            propertyKey="require_invoice_signature"
            labelElement={
              <SettingsLabel
                label={t('require_invoice_signature')}
                helpLabel={t('require_invoice_signature_help')}
              />
            }
            defaultValue={false}
          />
}
```

Without using these components and by placing the `<Checkbox />` component on each element where it is needed, along with adding label texts that also need to be adjusted separately, the code looks like this:

```
leftSide={
          <div className="flex">
            <Checkbox
              value={customHookForGettingValue()}
              onValueChange={(value) =>
                customHookForChangingValue('require_invoice_signature', value)
              }
            />
            
            <div
        className={classNames({
          'opacity-75': !checked && !isCompanySettingsActive,
          'cursor-pointer': !isCompanySettingsActive,
        })}
        onClick={() => {
          !isCompanySettingsActive && setChecked((current) => !current);
          !isCompanySettingsActive && onCheckboxChange?.(Boolean(!checked));
        }}
      >
            <div className="flex flex-col text-sm">
              <span className="font-medium" style={{ color: colors.$3 }}>
                {t('label')}
              </span>
              <span
                className="text-xs"
                style={{ color: colors.$3, opacity: 0.8 }}
              >
                {t('help_label')}
              </span>
            </div>
          </div>
          </div>
}
```

The code would look something like this, and this would be the case for each field within the Settings pages that are included in the `group/client settings`.

Of course, a custom hook would allow us to globalize the change of values within the Redux state using a checkbox. However, I believe that the component provides us with several additional benefits.

Firstly, it enhances code cleanliness and globalizes the parts that are evident. As an example, I can take the verification of whether the companySettings level is currently enabled, or perhaps, most importantly, it reduces the amount of code.

Another point, in my opinion, is that the checkbox in this part of the application is strictly used for one purpose, which is setting or removing properties from the Redux state. In my opinion, this is a compelling reason to create a custom component instead of using a regular Checkbox. This custom component can handle returning the component we actually need. For example, if we ever decide to use a `Toggle` instead of a `Checkbox`, the changes would be minimal, confined to just one component.

Additionally, we have cases where the `Checkbox` value needs to change when the text is clicked, whether on the label or helper label text. This requires additional logic within the main components.

The `PropertyCheckbox` component is well-prepared, and if there are any additional UI changes in the near future, such as adding icons next to the text, such modifications will be straightforward.

These are some of the most important reasons why I chose to use a component instead of a custom hook. However, I did this about seven days ago, and if I were to delve deeper into the logic, I'm sure I could think of several more reasons for this choice.

At the very beginning of creating the solution, I started without components that perform this unique task for all fields within the Settings fields. When I realized there was too much code and that a lot of it could be globalized, I changed the solution and did it this way. It was much easier for me to manipulate things in this manner with minimal changes.

If you look through the entire code and the changes to all the fields, everything is highly consistent and uniform. I believe this is the right case for such globalization because each field needs to perform the exact same task, with very rare exceptions on the Templates and Reminder page. However, even there, the component has been adapted for some edge cases.

To avoid making this text too long, this is my opinion. Of course, if you disagree with it, please let me know. I'm always open to better suggestions and solutions.

Let me know your thoughts.